src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -127,6 +159,34 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
     this
   }
 
+  /*
+   * Implemented by means of a buffer to keep the last n elements from being returned during iteration.
+   */
+  def dropRight(n: Int): Iterator[A] = {
```

Comment:
@Ichoran We could implement `takeRight` and `dropRight` in `Iterable`, where we could get two `Iterator`s. But I’m not sure we can, in general, make the assumption that it’s better to buffer rather than performing two traversals, nor the inverse…

I’m happy with the buffering solution.